File: day 02/Jakob - Typescript/Day2.ts
import { Command, Movement } from "./Command";

export {};

const fs = require("fs");
const text = fs.readFileSync("./input.txt", "utf-8");
const textByLine = text.split("\n");
const input: Command[] = textByLine.map((str) => new Command(str));

let horpos: number = 0;
let depth: number = 0;
let aim: number = 0;

for (let command of input) {
  switch (command.movement) {
    case Movement.DOWN: {
      aim += command.amount;
      break;
    }
    case Movement.UP: {
      aim -= command.amount;
      break;
    }
    case Movement.FORWARD: {
      horpos += command.amount;
      depth += command.amount * aim;
      if (depth < 0) {
        depth = 0;
      }
      break;
    }
  }
}

console.log("horizontal position: " + horpos);
console.log("depth: " + depth);
console.log("product: " + depth * horpos);
console.log("aim: " + aim);

File: day 02/Xavier - Python/day2.py
instructions = [(i.split(' ')[0], int(i.split(' ')[1])) for i in open('input.txt').readlines()]

from collections import defaultdict
totals = defaultdict(int)
for i in instructions:
    totals[i[0]] += i[1]
print(totals['forward']*(totals['down']-totals['up']))

forward = down = aim = 0
for i in instructions:
    if i[0] == 'forward':
        forward += i[1]
        down += aim*i[1]
    elif i[0] == 'down':
        aim += i[1]
    else:
        aim -= i[1]
print(forward * down)